step-framework-server/src/main/java/step/framework/server/CacheControlFilter.java

```diff
@@ -0,0 +1,38 @@
+package step.framework.server;
+
+import jakarta.servlet.Filter;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.FilterConfig;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.ServletRequest;
+import jakarta.servlet.ServletResponse;
+import jakarta.servlet.http.HttpServletResponse;
+import org.eclipse.jetty.server.Request;
+
+import java.io.IOException;
+
+public class CacheControlFilter implements Filter {
+
+    @Override
+    public void init(FilterConfig filterConfig) {}
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+        if (servletRequest instanceof Request) {
+            Request request = (Request) servletRequest;
+            HttpServletResponse httpResponse = (HttpServletResponse) response;
+
+            // Check if the request is for the index HTML file which is also implicit for the root URI context (welcome file)
+            if (request.getRequestURI().endsWith("index.html") || request.getPathInContext().equals("/")) {
+                // Set Cache-Control header to never cache HTML files
```

Comment:
The comment not exactly matches the if clause which applies only to index.html.

Reply:
@jeromecomte see my answer to next comment